core/controllers/custom_landing_pages.py

```diff
@@ -30,20 +30,29 @@ class FractionLandingRedirectPage(base.BaseHandler):
     @acl_decorators.open_access
     def get(self):
         """Handles GET requests."""
-        self.redirect('/learn/maths/fractions')
+        self.redirect('/maths/fractions')
+
+
+class TopicRedirectPage(base.BaseHandler):
+    """The handler redirecting the old landing page URL to the new one."""
+
+    @acl_decorators.open_access
+    def get(self, topic):
+        """Handles GET requests."""
+        self.redirect('/maths/%s' % topic)
 
 
 class TopicLandingPage(base.BaseHandler):
     """Page showing the topic landing page."""
 
     @acl_decorators.open_access
-    def get(self, subject, topic):
+    def get(self, topic):
         """Handles GET requests."""
-        if subject in feconf.AVAILABLE_LANDING_PAGES:
```

Comment:
Same reason for removal as for the classroom page. Since the modifications make it impossible for a URL with invalid subject to be passed in fot this handler, this code is redundant.

Reply:
See above.